test/src/test/java/org/corfudb/integration/LogReplicationIT.java

```diff
@@ -1225,26 +1222,23 @@ private LogReplicationSourceManager setupSourceManagerAndObservedValues(Set<Stri
 
         // Add this class as observer of the value of interest for the wait condition
         for (WAIT waitCondition : waitConditions) {
-            switch(waitCondition) {
-                case ON_ACK:
-                case ON_ACK_TS:
-                    ackMessages = sourceDataSender.getAckMessages();
-                    ackMessages.addObserver(this);
-                    break;
-                case ON_ERROR: // Wait on Error Notifications to Source
-                case ON_TIMEOUT_ERROR:
-                    errorsLogEntrySync = sourceDataSender.getErrors();
-                    errorsLogEntrySync.addObserver(this);
-                    break;
-                case ON_METADATA_RESPONSE:
-                    metadataResponseObservable = sourceDataSender.getMetadataResponses();
-                    metadataResponseObservable.addObserver(this);
-                case ON_SINK_RECEIVE: // Wait on Received Messages on Sink (Destination)
-                    sinkReceivedMessages = sourceDataSender.getSinkManager().getRxMessageCount();
-                    sinkReceivedMessages.addObserver(this);
-                default:
-                    // Nothing
-                    break;
+            if (waitCondition == WAIT.ON_ACK || waitCondition == WAIT.ON_ACK_TS) {
```

Comment:
if else is better than switch-case? Always thought it was the other way around.

Reply:
Yeah, It is. Unfortunately, in this case, it was resulting in the `switch condition without break` error for 2 `case` conditions. Hence I converted it to if-else.